### PR TITLE
Add ability to use puppet collections for this install

### PR DESCRIPTION
**Pull request by kostecky**

### CODE REVIEW

File: manifests/repository.pp [hunk 1]
```diff
@@ -10,19 +14,29 @@
...
-        enabled  => '1',
-      }
-      yumrepo { 'puppetlabs-products':
-        descr    => "Puppet Labs Products El ${::operatingsystemmajrelease} - \$basearch",
-        baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/products/\$basearch",
-        gpgcheck => '1',
-        gpgkey   => 'http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs',
-        enabled  => '1',
+      if $use_collections {
+        yumrepo { 'puppetlabs-PC1':
```
Comment (raphink) [inline manifests/repository.pp]: Why call it "PC1" if the collections version is a parameter?

Comment (kostecky) [inline manifests/repository.pp]: My Bad - one sec.

**Comment by raphink** [comment]
That looks like a good way to fix #18. I'll look into it further on Monday I think.

**Comment by kostecky** [comment]
@raphink Any chance to take a look at this?

**Comment by raphink** [comment]
@mcanevet can you have a look please?

**Comment by mcanevet** [comment]
@kostecky what about Debian?
We definitely some acceptance tests to prevent regression and prove that it works.

**Comment by dnaeon** [comment]
Any chance this could be fixed soon?

**Comment by raphink** [comment]
@dnaeon We stopped using this module recently. I will transfer the ownership to puppet-community soon most likely.

**Comment by dnaeon** [comment]
@raphink, thanks for the feedback.

Any idea which Puppet module can be used for managing the new Puppet Server? This one looked very promising, but since you are discontinuing support for it, do you know of any other modules that could be used as a replacement of this module?

Thanks,
Marin

**Comment by raphink** [comment]
@dnaeon We wouldn't have made this one if we didn't have a need for it, and we have chosen to stop managing the Puppet server stack with Puppet altogether.

**Comment by dnaeon** [comment]
@raphink thanks for the info! any chance you could share what you are using for managing the Puppet Server stack from now on?

**Comment by raphink** [comment]
@dnaeon We're using Docker. Our docker images are public on https://hub.docker.com/u/camptocamp. Hopefully, we'll be presenting a talk about how we compose them sometime soon (or a blog post, ping @mcanevet  :smile:)

**Comment by dnaeon** [comment]
Cool, thanks and looking forward to see any blogs/presentations about it :)

**Comment by raphink** [comment]
@dnaeon See https://twitter.com/raphink/status/664799221367853058

**Comment by bastelfreak** [comment]
HI @kostecky, this module recently moved to our namespace. Could you please to a rebase?

**Comment by vinzent** [comment]
IMHO the module should not provide a own class for the repo. the documentation should maybe refer to stahnma/puppetlabs_yum and ploperations/puppetlabs_apt . 

If anybody wants to install puppetserver with puppet one has either already configured Puppet repos or has all the rpm's in some internal repositories.

**Comment by dhollinger** [comment]
@vinzent I agree with that. Ultimately, this module probably shouldn't be handling the repositories for apt and yum. Those should probably be configured/managed at the profile level as a way to prevent duplicate resource declaration errors.

Example: A user wants to manage puppetdb, puppet agent, and puppetserver with Puppet and decide to declare the repository management in a profile, but that resource name conflicts with what's being set here?

Additionally, as @vinzent already pointed out, chances are that Puppet or Puppetserver has already been installed using the repos before this module ever gets applied, thus making the code a bit redundant.

**Comment by vinzent** [comment]
closing in favor of https://github.com/voxpupuli/puppet-puppetserver/issues/37

**Pull request closed without merging**